craft_parts/data/schema/parts.json

```diff
@@ -0,0 +1,423 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "definitions": {
+        "grammar-string": {
```

Comment:
Did we agree that the grammar would be processed before calling craft-parts?

Reply:
I know that this is a port, but personally I would prefer that craft-parts be purely a python object/dataclass definition [in the future] and leave any json/yaml schema validation to the application.  (and most of this PR would not be required).  If sanity checks are required for craft-parts, they perhaps can/should be done on init.

Reply:
grammar should not be part of craft-parts, it should receive processed ready to use entries

Reply:
@cjp256 I like the idea of having a parts dataclass to avoid dicts. We can consider refactoring it that way before 1.0 and the bulk of this change would be the `Part` refactoring you suggested earlier, but dealing with plugin schema definitions can be tricky. Any idea on how to handle them?